xwiki-platform-core/xwiki-platform-user/xwiki-platform-user-profile/xwiki-platform-user-profile-ui/src/main/resources/XWiki/XWikiUserPreferencesSheet.xml

```diff
@@ -287,7 +287,19 @@
       &lt;h2&gt;$escapetool.xml($services.localization.render('platform.core.profile.section.displayPreferences'))&lt;/h2&gt;
       &lt;dl&gt;
         #displayField('displayHiddenDocuments', 'platform.core.profile.displayHiddenDocuments', '0')
-        #displayField('accessibility'         , 'platform.core.profile.enableAccessibility'   , '0')
+      &lt;/dl&gt;
+      &lt;h2&gt;$escapetool.xml($services.localization.render('platform.core.profile.section.accessibilityPreferences'))&lt;/h2&gt;
+      &lt;dl&gt;
+        #set ($defaultValue = $xwiki.getXWikiPreference('font_size'))
+        #if ("$!defaultValue" == '')
+          #set ($defaultValue = 'Regular')
+        #end
+        #displayField('font_size', 'platform.core.profile.accessibility.fontSize', $defaultValue)
+        #set ($defaultValue = $xwiki.getXWikiPreference('underline'))
+        #if ("$!defaultValue" == '')
+          #set ($defaultValue = 'OnlyInlineLinks')
+        #end
+        #displayField('underline', 'platform.core.profile.accessibility.underline', $defaultValue)
```

Comment:
What I would suggest here is a) an initialization of the new settings if the legacy property is set and b) a hidden input to clear the legacy property.

Reply:
See https://github.com/xwiki/xwiki-platform/pull/2694#discussion_r1521821029 :+1: